Mathlib/Computability/Language.lean

```diff
@@ -0,0 +1,317 @@
+/-
+Copyright (c) 2020 Fox Thomson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fox Thomson
+
+! This file was ported from Lean 3 source module computability.language
+! leanprover-community/mathlib commit dd71334db81d0bd444af1ee339a29298bef40734
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Hom.Ring
+import Mathlib.Data.List.Join
+import Mathlib.Data.Set.Lattice
+
+/-!
+# Languages
+
+This file contains the definition and operations on formal languages over an alphabet. Note strings
+are implemented as lists over the alphabet.
+The operations in this file define a [Kleene algebra](https://en.wikipedia.org/wiki/Kleene_algebra)
+over the languages.
+-/
+
+
+open List Set
+
+universe v
+
+variable {α β γ : Type _}
+
+/-- A language is a set of strings over an alphabet. -/
+def Language (α) :=
+  Set (List α)
+  -- Porting note: Failed to derive the following, which seems necessary:
+  -- deriving Membership, Singleton, Insert, CompleteBooleanAlgebra
+#align language Language
+
+namespace Language
+
+variable {l m : Language α} {a b x : List α}
+
+-- Porting note: `reducible` attribute cannot be local.
+attribute [reducible] Language
+
+-- Porting note: it depends on 'EmptyCollection.emptyCollection',
+-- and it does not have executable code
+/-- Zero language has no elements. -/
+noncomputable instance : Zero (Language α) :=
+  ⟨(∅ : Set _)⟩
```

Comment:
```suggestion
instance : Zero (Language α) :=
  ⟨fun _ => False⟩
```
I don't really understand the error either, but this workaround works. Same goes for the other ones.